improver/utilities/spatial.py

```diff
@@ -268,8 +322,16 @@ def calculate_difference(cube: Cube, coord_name: str) -> ndarray:
             Array after the differences have been calculated along the
             specified axis.
         """
-        diff_axis = cube.coord_dims(coord_name)[0]
-        diff_along_axis = np.diff(cube.data, axis=diff_axis)
+        diff_axis = cube.coord(name_or_coord=coord_name)
+        diff_axis_number = cube.coord_dims(coord_name)[0]
+        diff_along_axis = np.diff(cube.data, axis=diff_axis_number)
+        if self._axis_wraps_around_meridian(diff_axis, cube):
+            first_column = cube.data[:, 0].reshape([-1, 1])
+            last_column = cube.data[:, -1].reshape([-1, 1])
+            wrap_around_diff = np.diff(
```

Comment:
This assumes that the cube is set up with x as rows and y as columns but this may not be true.
Can use an Improver module to force the cube to be x as rows and y as columns but have to make sure I've put it back again. Alternatively Might be able to work out correct index here on the fly.

Reply:
Have added a test for a cube using the opposite lat/long axis choices and fixed this.

Below is just my chain of thought/sanity check. Feel free to ignore.
...Wait... I produced the flipped code by transposing the cube. This flips which axis of cube.data each dim coord is assigned to, and also transposes the numpy array, so that the data represented by the cube is the same, but the longitudes now increase along the columns and the latitudes along the rows... which is exactly what I want. Okay, good.